libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -754,6 +754,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (auto const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += (constructorParams.empty() ? "" : ", ") + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+
+		Whiskers t(R"(
+			let size := datasize("<object>")
```

Comment:
You cannot use such generic variable names - this is not inside a newly generated function.

Reply:
For this, you also do not need a variable name, `datasize` is very cheap.